crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -582,20 +631,82 @@ pub struct Entities {
-    len: u32,
+
+    /// This functions exactly like [`Self::meta`], only it counts backwards instead of forwards.
+    remote_entities: Vec<EntityMeta>,
+    /// The counter for [`Self::remote_entities`]. This will equal it's length after flushing.
+    remote_reservations: RemoteEntityReserver,
+}
+
+/// Provides remote reservation of an entity from anywhere.
+#[derive(Debug, Clone)]
+pub struct RemoteEntityReserver(Arc<RemoteIdCursor>);
+
+impl RemoteEntityReserver {
+    /// We do this because [`Entity::PLACEHOLDER`] has index of `u32::MAX` and generation of 1.
+    /// By starting at 2, we ensure we don't conflict with that, and, we allow similar Entity constants to be made in the future.
+    const REMOTE_FIRST_GENERATION: NonZero<u32> = const {
+        // SAFETY: We pass 2, which is greater than 0.
+        unsafe { NonZero::new_unchecked(2) }
+    };
+
+    const REMOTE_FIRST_META: EntityMeta = const {
+        let mut default = EntityMeta::EMPTY;
+        default.generation = Self::REMOTE_FIRST_GENERATION;
+        default
+    };
+
+    /// Reserves an entity that will be allocated in [`Entities::flush`].
+    #[expect(
+        clippy::allow_attributes,
+        reason = "`clippy::unnecessary_fallible_conversions` may not always lint."
+    )]
+    #[allow(
+        clippy::unnecessary_fallible_conversions,
+        reason = "`RemoteIdCursor` may already be a u32."
+    )]
+    pub fn reserve(&self) -> Entity {
+        let index_from_top = self.0.fetch_add(1, Ordering::Relaxed);
+        let index_from_top = u32::try_from(index_from_top).expect("Too many entities.");
+        let index = u32::MAX - index_from_top;
+        Entity::from_raw_and_generation(index, Self::REMOTE_FIRST_GENERATION)
+    }
+
+    /// returns true if this index is from a remote reservation.
+    pub fn is_index_from_remote(&self, index: u32) -> bool {
+        self.0.load(Ordering::Relaxed) > (u32::MAX - index) as RemoteInner
+    }
+
+    /// Reserves `count` entities that will be allocated in [`Entities::flush`].
+    pub fn reserve_entities(&self, count: u32) -> RemoteReserveEntitiesIterator {
+        let list_index_start = self.0.fetch_add(count as RemoteInner, Ordering::Relaxed);
+        let list_index_start = u32::try_from(list_index_start).expect("Too many entities.");
+        let index_end = u32::MAX - list_index_start;
+        let index_start = index_end - count;
+        RemoteReserveEntitiesIterator {
+            new_indices: index_start..index_end,
+        }
+    }
 }
 
 impl Entities {
-    pub(crate) const fn new() -> Self {
+    pub(crate) fn new() -> Self {
         Entities {
             meta: Vec::new(),
             pending: Vec::new(),
             free_cursor: AtomicIdCursor::new(0),
-            len: 0,
+            remote_entities: Vec::new(),
+            remote_reservations: RemoteEntityReserver(Arc::new(RemoteIdCursor::new(0))),
         }
     }
 
+    /// Constructs a new [`RemoteEntityReserver`] for this [`Entities`].
```

Comment:
```suggestion
    /// Get a [`RemoteEntityReserver`] for this [`Entities`].
```

The previous description sounded like it's creating like a new allocator or something to me.